### DISCUSSION TITLE
API For Lighthouse Score Calculator

### DISCUSSION THREAD

**Discussion by kobearv** [GoogleChrome/lighthouse#15622] [Q&A]
I have all the 6 web vital values that are used to calculate the performance score. Since I don't know the corresponding metric scores lighthouse uses I cannot calculate the performance score on my own. Is there an API call I can use in code where I can add these web vital values and get a response with the performance score?

**Comment by adamraine** [comment]
We don't really have a publicly exposed API for this unfortunately.

As an example, this is how LCP computes it's 0-1 score: https://github.com/GoogleChrome/lighthouse/blob/main/core/audits/metrics/largest-contentful-paint.js#L82

And this is how the 0-1 scores of all metrics are used to compute the overall 0-1 performance score: https://github.com/GoogleChrome/lighthouse/blob/main/core/scoring.js#L73

**Comment by connorjclark** [comment]
This may be of use to you: https://github.com/paulirish/lh-scorecalc/tree/master/script

**Reply by kobearv** [reply]
I did use this and add the vital values as parameters in a get query, but the response didn't return the scores

**Reply by connorjclark** [reply]
This isn't an API service. The code is there for your use, though.

**Reply by kobearv** [reply]
Got it! So my problem now is that i called the generateReport method with a lighthouse run object which had a performance score and the web vitals needed, but when I get the report back, the main html page of the report shows that performance score(the one in the lighthouse run object) but the score in the calculator is way off.  For example, let's say I arrived at a performance score of 0.72(got after averaging performance scores from 2 lighthouse runs with performance scores of 0.49 and 0.94) and then this was added to the lhr object along with the averaged scores of the 6 web vitals. The main html page of the report shows 72 but the lh score calculator (after clicking on the link) shows 60. I'm guessing it's because the metric scores need to be considered.
